internal/node/controller/auth.go

```diff
@@ -103,3 +109,186 @@ func getUserCertAndInfo(controller NodeController, username string) (*x509.Certi
 
 	return cert, user, nil
 }
+
+// BFF Auth - Allow for Habitat node's to authenticate with each other.
+
+type BFFChallengeRequest struct {
+	DID string `json:"did"`
+}
+
+type FriendStore map[string]*Friend
+
+type Friend struct {
+	DID       string `json:"did"`
+	PublicKey crypto.PublicKey
+}
+
+type FriendRequest struct {
+	DID                string `json:"did"`
+	PublicKeyMultibase string `json:"public_key_multibase"`
+}
+
+type BFFProvider struct {
+	challengePersister bffauth.ChallengeSessionPersister
+	friends            FriendStore
+	signingKey         []byte
+}
+
+func NewBFFProvider(challengePersister bffauth.ChallengeSessionPersister, friends FriendStore, signingKey []byte) *BFFProvider {
+	return &BFFProvider{
+		challengePersister: challengePersister,
+		friends:            friends,
+		signingKey:         signingKey,
+	}
+}
+
+func (p *BFFProvider) GetRoutes() []api.Route {
+	return []api.Route{
+		newRoute(http.MethodPost, "/node/bff/challenge", p.handleChallenge),
+		newRoute(http.MethodPost, "/node/bff/auth", p.handleAuth),
+		newRoute(http.MethodPost, "/node/bff/add_friend", p.handleAddFriend),
+		newRoute(http.MethodGet, "/node/bff/test", p.handleTest),
+	}
+}
+
+func (p *BFFProvider) handleChallenge(w http.ResponseWriter, r *http.Request) {
+	var req BFFChallengeRequest
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		http.Error(w, "Invalid request body", http.StatusBadRequest)
+		return
+	}
```

Comment:
we should standardize on if we're using `json.NewDecoder` vs. `json.Unmarshal` ... i prefer the latter if we're not expecting the input to be a huge stream

Reply:
Fine to not do in this PR, I'll standardize this as a follow up